sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/DiskEncryptionSet.java

```diff
@@ -0,0 +1,236 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.resourcemanager.compute.models;
+
+import com.azure.core.annotation.Fluent;
+import com.azure.resourcemanager.authorization.models.BuiltInRole;
+import com.azure.resourcemanager.compute.ComputeManager;
+import com.azure.resourcemanager.compute.fluent.models.DiskEncryptionSetInner;
+import com.azure.resourcemanager.resources.fluentcore.arm.models.GroupableResource;
+import com.azure.resourcemanager.resources.fluentcore.model.Appliable;
+import com.azure.resourcemanager.resources.fluentcore.model.Creatable;
+import com.azure.resourcemanager.resources.fluentcore.model.Refreshable;
+import com.azure.resourcemanager.resources.fluentcore.model.Updatable;
+
+/** An immutable client-side representation of an Azure disk encryption set. */
+@Fluent
+public interface DiskEncryptionSet
+    extends GroupableResource<ComputeManager, DiskEncryptionSetInner>,
+        Updatable<DiskEncryptionSet.Update>,
+        Refreshable<DiskEncryptionSet> {
+    /** @return resource id of the Azure key vault containing the key or secret */
+    String keyVaultId();
+
+    /** @return id representing the key in KeyVault */
+    String keyId();
+
+    /** @return the System Assigned (Local) Managed Service Identity specific Active Directory service principal ID
+     *          assigned to the disk encryption set.
```

Comment:
I am not sure about what is the `(Local)` here? Do we have a `Remote`?

Reply:
No, currently we only have system assigned. I'll remove it for now.